### ISSUE TITLE
Change separator in link rule references to `:`

### ISSUE THREAD

**Issue by igordejanovic** [textX/textX#384]
Usage of `|` to separate rule to link to from the rule used to parse the reference [leads to confusion with ordered choice](https://stackoverflow.com/questions/73518617/textx-state-machine-model-not-getting-parsed-properly). This has already been noticed by Langium team and [they choose `:`](https://langium.org/docs/grammar-language/#cross-references).

This change can be done in a backward compatible manner as allowing both `:` and `|` while noting in the docs that `|` is deprecated and removing it completely in two major versions ahead.

**Comment by goto40** [comment]
Great. Do we have some kind of cooperation with them? I told them about our rrel last month.

(I am playing with langium at the moment...)

**Comment by igordejanovic** [comment]
We don't as far as I can tell but since we are working on a very similar meta-language it is good to share ideas :). I recall that @msujew noted during a Strumenta virtual meetup that they changed the separator as their users reported an often confusion with ordered choice.

**Comment by ikondov** [comment]
Recently, I have started a (not yet released) project using a grammar with many link rule references (containing also RREL). I would appreciate changing the `|` with a `:` if possible.

**Comment by igordejanovic** [comment]
Implementing `:` as a separator is easy to do, it can be backward compatible if we retain old separator as an option.

What bothers me a bit is that we are already using `:` in RREL prefixes. So, if prefix is used it looks a bit awkward. For example:

```
type=[Type:ID:+m:~active_types.types]
```

There are two solutions:
1. Choosing a different seaparator. Ideally, not used in RREL and the rest of the grammar language.
2. Using different separators at first and second position.

At first I thought second solution is not an option but after giving it a thought for awhile I'm now more in its favour.

For example:
```
type=[Type:ID|+m:~active_types.types]
```

Separator `|` can hardly be a source of confusion now as RREL is clearly not a rule reference.

It would be nicer for leaving the second part out (we don't support that at the moment):
```
type=[Type|+m:~active_types.types]
```

While previously we should probably do:
```
type=[Type||+m:~active_types.types]
```

What do you think? Should we choose a completely different separator (which one?) or use `:` at first position (between target rule and parser rule) and `|` at second position (before RREL).

**Comment by goto40** [comment]
Solution 2 looks good to me (`:` first, then `|`) . I think a basic compatibility to langium is a good thing.

**Comment by igordejanovic** [comment]
This is implemented in #385 and merged to master. It will be a part of the next minor release.
